src/android/toga_android/widgets/datepicker.py

```diff
@@ -0,0 +1,103 @@
+from datetime import date, datetime
+
+from ..libs.android import R__drawable
+from ..libs.android.view import OnClickListener, View__MeasureSpec
+from ..libs.android.widget import DatePickerDialog
+from ..libs.android.widget import \
+    DatePickerDialog__OnDateSetListener as OnDateSetListener
+from ..libs.android.widget import EditText
+from .base import Widget
+
+
+def _date_to_milli(value):
+    if isinstance(value, str):
+        value = date.fromisoformat(value)
+    datetime_value = datetime.combine(value, datetime.min.time())
+    timestamp = datetime_value.timestamp()
+    return int(timestamp * 1000)
+
+class TogaDatePickerClickListener(OnClickListener):
+    def __init__(self, picker_impl):
+        super().__init__()
+        self.picker_impl = picker_impl
+    
+    
+    def onClick(self, _):
+        self.picker_impl._create_dialog()
+
+
+class TogaDatePickerDateSetListener(OnDateSetListener):
+    def __init__(self, picker_impl):
+        super().__init__()
+        self.picker_impl = picker_impl
+
+    def onDateSet(self, _, year, month, day):
+        new_value = date(year, month + 1, day)
+        
+        self.picker_impl._showing = False
+        self.picker_impl._dialog = None
+        self.picker_impl.set_value(new_value)
+        if self.picker_impl.interface.on_change:
+            self.picker_impl.interface.on_change(self.picker_impl, new_value)
```

Comment:
The value shouldn't be passed as part of the handler.

Reply:
Oh, alright, that is of corrected now